base/jupyter_server_config.py

```diff
@@ -0,0 +1,59 @@
+# Copyright (c) Jupyter Development Team.
+# Distributed under the terms of the Modified BSD License.
+# mypy: ignore-errors
+import os
+import stat
+import subprocess
+from pathlib import Path
+
+from jupyter_core.paths import jupyter_data_dir
+
+c = get_config()  # noqa: F821
+c.ServerApp.ip = "0.0.0.0"
+c.ServerApp.open_browser = False
+
+# to output both image/svg+xml and application/pdf plot formats in the notebook file
+c.InlineBackend.figure_formats = {"png", "jpeg", "svg", "pdf"}
+
+# https://github.com/jupyter/notebook/issues/3130
+c.FileContentsManager.delete_to_trash = False
+
+# Generate a self-signed certificate
+OPENSSL_CONFIG = """\
+[req]
+distinguished_name = req_distinguished_name
+[req_distinguished_name]
+"""
+if "GEN_CERT" in os.environ:
+    dir_name = Path(jupyter_data_dir())
+    dir_name.mkdir(parents=True, exist_ok=True)
+    pem_file = dir_name / "notebook.pem"
+
+    # Generate an openssl.cnf file to set the distinguished name
+    cnf_file = Path(os.getenv("CONDA_DIR", "/usr/lib")) / "ssl/openssl.cnf"
+    if not cnf_file.exists():
+        cnf_file.write_text(OPENSSL_CONFIG)
+
+    # Generate a certificate if one doesn't exist on disk
+    subprocess.check_call(
+        [
+            "openssl",
+            "req",
+            "-new",
+            "-newkey=rsa:2048",
+            "-days=365",
+            "-nodes",
+            "-x509",
+            "-subj=/C=XX/ST=XX/L=XX/O=generated/CN=generated",
+            f"-keyout={pem_file}",
+            f"-out={pem_file}",
+        ]
+    )
+    # Restrict access to the file
+    pem_file.chmod(stat.S_IRUSR | stat.S_IWUSR)
+    c.ServerApp.certfile = str(pem_file)
+
+# Change default umask for all subprocesses of the Server if set in the environment
+if "NB_UMASK" in os.environ:
+    os.umask(int(os.environ["NB_UMASK"], 8))
+
```

Comment:
As I suggested before, what if we loaded other partial config python files from a directory (like we do in the zero2jh python config)? It would make things cleaner as we wouldn't need to `echo` from our startup bash scripts.

In the bash script we load files with the extension `.sh` and here files with `.py`(or we just create a new folder where these files would live, split from the sh ones).
I would also add the possibility of putting those files under subfolders (or we store them in subfolders, but then merge them when we ADD).

I would even push this as an upstream contribution...

Reply:
It's just a couple of lines at the end of the file. So we can just easily put a comment saying that this is SWAN-specific.

Reply:
But iiuc you also want to move .sh and .py into subdirs?

Reply:
We don't need that. It was a suggestion. So let's keep them in a single folder and just add the py part.

Reply:
Or we just keep the py files split and we append them via a command issued in the sh file.

Reply:
We decided to inject our config py with a bash script, so we don't need to automatically import it here.